### PR TITLE
Instructions for updating from version 40 - 43

### PR DESCRIPTION
**Pull request by gaurav-nelson**
Based on content from https://help.stackrox.com/docs/upgrade-stackrox/from-40/

Preview: https://openshift-docs-git-upgrade-from-40-gnelson.vercel.app/openshift-acs/master/upgrading/upgrading_roxctl/upgrade-from-40-43.html

[bfuru]: I edited the preview URL s/.adoc/.html so it would render.

### CODE REVIEW

File: upgrading/upgrading_roxctl/upgrade-from-40-43.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
+[id="upgrade-from-40-43"]
+= Upgrading from version 3.0.40 till 3.0.43
+include::modules/common-attributes.adoc[]
+:context: upgrade-from-40-43
+
+toc::[]
+
+[role="_abstract"]
+
+Upgrade to the latest version of Red Hat Advanced Cluster Security for Kubernetes from version 3.0.44 or higher.
```
Comment (bobfuru) [inline upgrading/upgrading_roxctl/upgrade-from-40-43.adoc:10]: This sentence is confusing for a few reasons: 
-  I think "till" --> "through" as in the original docs makes more sense in the title. 
- The title says upgrading "from 3.0.40 till 3.0.43" and the intro sentence says "from 3.0.44". I think in this file that line 10 should say "from 3.0.40 through 3.0.43" to match the title.
- We don't generally precede our versions numbers with the word "version". ISG has some guidance about this:
  > At the first occurrence in text, it is preferable to spell out the full official name of the product and the version...In running text, it is acceptable to use the word version, lowercase, to make a generic reference to a product name that has already been mentioned.

  Thus, it's long-winded but I'd suggest revising the assembly titles to "Upgrading from Red Hat Advanced Cluster Security for Kubernetes 3.0.40 through 3.0.43" and "Upgrading Red Hat Advanced Cluster Security for Kubernetes 3.0.44 or higher" and changing dropping the word "version" when it precedes a number.

File: upgrading/upgrading_roxctl/upgrade-from-40-43.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+To upgrade Red Hat Advanced Cluster Security for Kubernetes to the latest version, you must perform the following:
```
Comment (bobfuru) [inline upgrading/upgrading_roxctl/upgrade-from-40-43.adoc:12]: I'd say "you must perform the following steps" (or actions)

File: upgrading/upgrading_roxctl/upgrade-from-40-43.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+include::modules/back-up-central-database.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+* xref:../../cli/getting-started-cli.adoc#cli-authentication_cli-getting-started[Authenticating using the roxctl CLI]
+
+[id="upgrade-central-cluster"]
+== Upgrading the Central cluster
+
+After you have backed up Central database, the next step is to upgrade Central and Scanner.
```
Comment (bobfuru) [inline upgrading/upgrading_roxctl/upgrade-from-40-43.adoc:27]: s/up Central/up the Central

**Review by bobfuru — COMMENTED** [review]
A couple of minor comments to address before merge. Thanks!

File: modules/upgrade-central-40.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+[id="upgrade-central-40_{context}"]
+= Upgrading Central
+
+You can update Central to the latest version by downloading and deploying the updated images.
+
+.Prerequisites
+
+* If you deploy images from a private image registry, first push the new image into your private registry, and then replace your image registry for the commands in this section.
+* If you used Red Hat UBI-based images when you installed {product-title}, replace the image names for the commands in this section with the following UBI-based image names:
+** For Central, Sensor, and Compliance use `\stackrox.io/main-rhel`
```
Comment (bobfuru) [inline modules/upgrade-central-40.adoc:14]: s/Compliance use/Compliance, use
Also just confirming that `stackrox` is still applicable in these URLs

File: modules/upgrade-central-40.adoc [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+** For Scanner use `\stackrox.io/scanner-rhel` and `\stackrox.io/scanner-db-rhel`
+** For Collector use `\collector.stackrox.io/collector-rhel`
```
Comment (bobfuru) [inline modules/upgrade-central-40.adoc:16]: Also commas after Scanner and Collector

File: modules/upgrade-central-40.adoc [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+.Procedure
+
+. Run the following commands to upgrade Central:
++
+[source,terminal]
+----
+oc -n stackrox patch deploy/central -p '{"spec":{"template":{"metadata":{"annotations":{"traffic.sidecar.istio.io/excludeInboundPorts":"8443"}}}}}'
```
Comment (bobfuru) [inline modules/upgrade-central-40.adoc:24]: Missing a `$` prompt here

File: modules/upgrade-scanner-40.adoc [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+You can update Scanner to the latest version by using the `roxctl` CLI.
+
+.Prerequisites
+
+* If you deploy images from a private image registry, first push the new image into your private registry, and then replace your image registry for the commands in this section.
+* If you used Red Hat UBI-based images when you installed {product-title}, replace the image names for the commands in this section with the following UBI-based image names:
+** For Central, Sensor, and Compliance use `\stackrox.io/main-rhel`
+** For Scanner use `\stackrox.io/scanner-rhel` and `\stackrox.io/scanner-db-rhel`
+** For Collector use `\collector.stackrox.io/collector-rhel`
```
Comment (bobfuru) [inline modules/upgrade-scanner-40.adoc:16]: Same comment as above about placing a comma before "use"

File: modules/upgrade-scanner-40.adoc [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+[source,terminal]
+----
+$ oc -n stackrox patch deploy/scanner -p '{"spec":{"template":{"spec":{"containers":[{"name":"scanner","command":["/entrypoint.sh"],"volumeMounts":[{"mountPath":"/etc/ssl","name":"central-etc-ssl-volume"},{"mountPath":"/usr/local/share/ca-certificates/","name":"additional-ca-volume","readOnly":true}]}],"volumes":[{"name":"additional-ca-volume","secret":{"defaultMode":420,"optional":true,"secretName":"additional-ca"}},{"emptyDir":{},"name":"central-etc-ssl-volume"}]}}}}'
+----
++
+[source,terminal]
+----
+$ oc -n stackrox patch deploy/scanner -p '{"spec":{"template":{"metadata":{"annotations":{"traffic.sidecar.istio.io/excludeInboundPorts":"8080,8443"}}}}}'
+----
+. Apply the patch for Scanner database:
```
Comment (bobfuru) [inline modules/upgrade-scanner-40.adoc:31]: the Scanner database

File: modules/upgrade-scanner-40.adoc [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
++
+[source,terminal]
+----
+$ oc -n stackrox patch deploy/scanner-db -p '{"spec":{"template":{"metadata":{"annotations":{"traffic.sidecar.istio.io/excludeInboundPorts":"5432"}}}}}'
+----
+. Apply the patch for Scanner service:
```
Comment (bobfuru) [inline modules/upgrade-scanner-40.adoc:37]: the Scanner service

File: modules/upgrade-scanner-40.adoc [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
++
+[source,terminal]
+----
+$ oc -n stackrox patch svc/scanner -p '{"spec":{"ports":[{"port":8443,"name":"grpcs-scanner"}]}}'
+----
+. Apply the patch for Scanner database service:
```
Comment (bobfuru) [inline modules/upgrade-scanner-40.adoc:43]: the Scanner

File: modules/upgrade-scanner-40.adoc [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
++
+[source,terminal]
+----
+$ oc -n stackrox patch svc/scanner-db -p '{"spec":{"ports":[{"port":5432,"name":"db"}]}}'
+----
+. Apply the patch for Scanner Horizontal Pod Autoscaler.
```
Comment (bobfuru) [inline modules/upgrade-scanner-40.adoc:49]: the Scanner

**Review by bobfuru — COMMENTED** [review]
Mostly minor style suggestions. Not sure if adding an article "the" should apply in all instances that I've noted; up to you. 😁

File: modules/back-up-central-database.adoc [hunk 12]
```diff
@@ -16,8 +16,15 @@ You can back up Central's database and use that backup for rolling back from a f
 .Procedure
 
 * Run the backup command:
+** For {product-title} version 3.0.55 and newer:
```
[the remark this answers is not in the document]
Comment (bobfuru) [reply modules/back-up-central-database.adoc:19]: ^ Feel free to ignore this comment if it is due to reusing this module.

File: modules/upgrade-central-40.adoc (hunk 4 above)
Comment (gaurav-nelson) [reply modules/upgrade-central-40.adoc:14]: Yeah its still `stackrox`, they still have to switch names.

File: modules/back-up-central-database.adoc [hunk 13]
```diff
@@ -16,8 +16,15 @@ You can back up Central's database and use that backup for rolling back from a f
...
 +
 [source,terminal]
 ----
 $ roxctl -e "$CENTRAL_ADDRESS" central backup
 ----
+** For {product-title} version 3.0.54 and olders:
```
[the remark this answers is not in the document]
Comment (gaurav-nelson) [reply modules/back-up-central-database.adoc]: @bobfuru 

Heads up! I've changed all instances of `{product-title} version x.x.xx` to `{product-title} x.x.xx` across multiple files which were not part of the original PR you reviewed.

[the remark this answers is not in the document]
Comment (bobfuru) [reply modules/back-up-central-database.adoc]: Thanks!

**Review by bobfuru — COMMENTED** [review]
LGTM!

**Pull request merged**